packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/ModalAddAlias.tsx

```diff
@@ -0,0 +1,273 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import {
+  ODS_THEME_COLOR_INTENT,
+  ODS_THEME_TYPOGRAPHY_SIZE,
+  ODS_THEME_TYPOGRAPHY_LEVEL,
+  ODS_THEME_COLOR_HUE,
+} from '@ovhcloud/ods-common-theming';
+import {
+  OsdsFormField,
+  OsdsInput,
+  OsdsSelect,
+  OsdsSelectOption,
+  OsdsText,
+} from '@ovhcloud/ods-components/react';
+import { ODS_INPUT_SIZE, ODS_INPUT_TYPE } from '@ovhcloud/ods-components';
+import { useNotifications } from '@ovhcloud/manager-components';
+import { useQuery } from '@tanstack/react-query';
+import { useDomains, useGenerateUrl, usePlatform } from '@/hooks';
+import Modal from '@/components/Modals/Modal';
+
+import {
+  getZimbraPlatformAccountDetail,
+  getZimbraPlatformAccountDetailQueryKey,
+} from '@/api/account';
+import { postZimbraPlatformAlias } from '@/api/alias';
+
+export default function ModalAddAndEditOrganization() {
+  const { t } = useTranslation('accounts/alias/add');
+  const { platformId } = usePlatform();
+  const [searchParams] = useSearchParams();
+  const editEmailAccountId = searchParams.get('editEmailAccountId');
+  const [isLoading, setIsLoading] = useState(true);
+  const { addError, addSuccess } = useNotifications();
+  const navigate = useNavigate();
+  const goBackUrl = useGenerateUrl('..', 'path', { editEmailAccountId });
+  const goBack = () => navigate(goBackUrl);
+
+  type FieldType = {
+    value: string;
+    hasError?: boolean;
+    required?: boolean;
+  };
+
+  interface FormTypeInterface {
+    [key: string]: FieldType;
+  }
+
+  interface FormInputRegexInterface {
+    [key: string]: RegExp;
+  }
+
+  const formInputRegex: FormInputRegexInterface = {
```

Comment:
**nitpick:** This should be moved to constant file.

Reply:
It's applied